skcosmo/pcovr/pcovr.py

```diff
@@ -1,76 +1,239 @@
-    :type space: {'feature', 'structure', 'auto'}
+    explained_variance_ : ndarray of shape (n_components,)
+        The amount of variance explained by each of the selected components.
 
-    :param lr_args: dictionary of arguments to pass to the Ridge Regression
-                    in estimating :math:`{\\mathbf{\\hat{Y}}}`
+        Equal to n_components largest eigenvalues
+        of the PCovR-modified covariance matrix of :math:`\mathbf{X}`.
+
+    singular_values_ : ndarray of shape (n_components,)
+        The singular values corresponding to each of the selected components.
 
     References
+    ----------
         1.  S. de Jong, H. A. L. Kiers, 'Principal Covariates
             Regression: Part I. Theory', Chemometrics and Intelligent
             Laboratory Systems 14(1): 155-164, 1992
         2.  M. Vervolet, H. A. L. Kiers, W. Noortgate, E. Ceulemans,
             'PCovR: An R Package for Principal Covariates Regression',
             Journal of Statistical Software 65(1):1-14, 2015
+        3.  B. A. Helfrecht, R. K. Cersonsky, G. Fraux, and M. Ceriotti,
+            'Structure-property maps with Kernel principal covariates regression',
+            Machine Learning: Science and Technology 1(4):045021, 2020
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> from skcosmo.pcovr import PCovR
+    >>> X = np.array([[-1, 1, -3, 1], [1, -2, 1, 2], [-2, 0, -2, -2], [1, 0, 2, -1]])
+    >>> Y = np.array([[ 0, -5], [-1, 1], [1, -5], [-3, 2]])
+    >>> pcovr = PCovR(mixing=0.1, n_components=2)
+    >>> pcovr.fit(X)
+    PCovR(lr_args=None, mixing=0.1, n_components=2, space=None, tol=None)
+    >>> T = pcovr.transform(X)
+        [[-2.48017109 -1.54378072]
+         [ 2.74724894 -1.67904456]
+         [-2.56778727  1.44091806]
+         [ 2.30070942  1.78190722]]
+    >>> Yp = pcovr.predict(X)
+        [[ 0.03200724, -5.01754987],
+         [-1.05781286,  1.14547488],
+         [ 0.95713937, -4.96456897],
+         [-2.99288563,  1.95975794]]
     """
 
     def __init__(
         self,
         mixing=0.0,
         n_components=None,
-        regularization=1e-6,
+        *,
+        regularization=None,
         tol=1e-12,
-        space=None,
-        lr_args=dict(alpha=1e-6, fit_intercept=False, tol=1e-12),
+        space="auto",
+        estimator=LR(alpha=1e-6, fit_intercept=False, tol=1e-12),
     ):
 
         self.mixing = mixing
+        self.n_components = n_components
         self.regularization = regularization
         self.tol = tol
         self.space = space
-        self.lr_args = lr_args
-        self.n_components = n_components
+
         self.whiten = False
-        self._eig_solver = partial(
-            eig_solver, n_components=self.n_components, tol=self.tol, add_null=True
-        )
+        self.estimator = estimator
+
+        if regularization is None:
+            self.regularization = getattr(self.estimator, "alpha_", 1e-6)
```

Comment:
codecov say a test for this case is missing

Reply:
Fixed!